### PR TITLE
Audit log: Optionally mute background jobs (internal actors). 

### PR DESCRIPTION
**Pull request by vrto**
## Description

Site admins can now optionally enable detailed (noisy) audit logs for the background jobs. For the sake of simplicity, any action that uses _internal actor_ will be considered a "background job" in the audit log context.

```
log.auditLog.backgroundJobs = true/false
```

The default value is `false.`

This PR also refactors the `auditLogEnabled` logic, which has been repeated at a few places, and groups that logic into a single location, accessible via [`audit.IsEnabled(cfg, setting)`](https://sourcegraph.sourcegraph.com/-/editor?remote_url=https://github.com/sourcegraph/sourcegraph.git&branch=mv/auditlog/mute-background-jobs&file=internal/audit/audit.go&start_row=86&start_col=5&end_row=86&end_col=5&editor=JetBrains&version=v2.0.1) function.

## Test plan

- Automated tests (both new, and refactored)
- Manual testing, see the attached gifs for turning the audit logs on and off

### CODE REVIEW

File: internal/audit/audit.go [hunk 1]
```diff
@@ -8,24 +8,26 @@ import (
...
-
-	client := requestclient.FromContext(ctx)
 	act := actor.FromContext(ctx)
 
-	//TODO make this configurable
-	if act.Internal {
+	// internal actors add a lot of noise to the audit log
+	if act.Internal && !IsEnabled(conf.SiteConfig(), BackgroundJobs) {
 		return
 	}
```
Comment (bobheadxi) [inline internal/audit/audit.go:25]: IIRC it's not just background jobs that uses internal actors right? Maybe `InternalTraffic` is a more apt "abstraction" for the configuration?

File: CHANGELOG.md [hunk 2]
```diff
@@ -30,7 +30,8 @@ All notable changes to Sourcegraph are documented in this file.
 - Git server access logs are now compliant with the audit logging format. Breaking change: The 'actor' field is now nested under 'audit' field. [#41865](https://github.com/sourcegraph/sourcegraph/pull/41865)
 - All Perforce rules are now stored together in one column and evaluated on a "last rule takes precedence" basis. [#41785](https://github.com/sourcegraph/sourcegraph/pull/41785)
 - Security events are now a part of the audit log. [#42653](https://github.com/sourcegraph/sourcegraph/pull/42653)
-- "GC AUTO" is now the default garbage collection job. We disable sg maintenance, which had previously replace "GC AUTO", after repeated reports about repo corruption. [#42856](https://github.com/sourcegraph/sourcegraph/pull/42856) 
+- "GC AUTO" is now the default garbage collection job. We disable sg maintenance, which had previously replace "GC AUTO", after repeated reports about repo corruption. [#42856](https://github.com/sourcegraph/sourcegraph/pull/42856)
```
Comment (mrnugget) [inline CHANGELOG.md:33]: What's the diff here? 👀

File: internal/audit/audit.go [hunk 3]
```diff
@@ -71,3 +73,30 @@ type Record struct {
 	// Fields hold any additional context relevant to the Action
 	Fields []log.Field
 }
+
+type AuditLogSetting = int
+
+const (
+	GitserverAccess = iota
```
Comment (mrnugget) [inline internal/audit/audit.go:80]: This makes it the proper type:

```suggestion
	GitserverAccess AuditLogSetting  = iota
```

**Review by mrnugget — APPROVED** [review]

File: internal/audit/audit.go (hunk 3 above)
Comment (mrnugget) [reply internal/audit/audit.go:80]: I guess it doesn't matter? Learned something about iota again.

File: internal/audit/audit.go (hunk 1 above)
Comment (vrto) [reply internal/audit/audit.go:25]: @bobheadxi sure, happy to change it.

File: CHANGELOG.md (hunk 2 above)
Comment (vrto) [reply CHANGELOG.md:33]: GitHub is showing weird diff here, [these are the added](https://github.com/sourcegraph/sourcegraph/pull/42946/files/78f30a0ac50af987447eedf24ccf1aabe6247d3d#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR34) lines

File: internal/audit/audit.go (hunk 3 above)
Comment (vrto) [reply internal/audit/audit.go:80]: Yeah, it doesn't :)

**Pull request merged**